dcos_test_utils/ssh_client.py

```diff
@@ -136,3 +132,133 @@ def wait_for_ssh_connection(self, host: str, port: int=22) -> None:
 
     def add_ssh_user_to_docker_users(self, host: str, port: int=22):
         self.command(host, ['sudo', 'usermod', '-aG', 'docker', self.user], port=port)
+
+
+@contextmanager
+def make_slave_pty():
+    master_pty, slave_pty = pty.openpty()
+    yield slave_pty
+    os.close(slave_pty)
+    os.close(master_pty)
+
+
+def parse_ip(ip: str) -> (str, int):
+    """  takes an IP string and either a hostname and either the given port or
+    the default ssh port of 22
+    """
+    tmp = ip.split(':')
+    if len(tmp) == 2:
+        return tmp[0], int(tmp[1])
+    elif len(tmp) == 1:
+        # no port, assume default SSH
+        return ip, 22
+    else:
+        raise ValueError(
+            "Expected a string of form <ip> or <ip>:<port> but found a string with more than one " +
+            "colon in it. NOTE: IPv6 is not supported at this time. Got: {}".format(ip))
+
+
+class MultiRunner(SshClient):
+    def __init__(
+            self,
+            user: str,
+            key: str,
+            targets: list,
+            process_timeout=120,
+            parallelism=10):
+        super().__init__(user, key)
+        self.process_timeout = process_timeout
+        self.__targets = targets
+        self.__parallelism = parallelism
+
+    @asyncio.coroutine
+    def run_cmd_return_dict_async(self, cmd: list) -> dict:
+        """
+        Runs cmd with a pTTY and times out if necessary
+        Returns a dict of data about the process
+        """
+        with make_slave_pty() as slave_pty:
+            process = yield from asyncio.create_subprocess_exec(
```

Comment:
`yield from` and `@asyncio.coroutine` decorator is old syntax. In 3.6, it is deprecated. We could still use this while we are on 3.5.
In 3.6, instead of `@asyncio.coroutine`, this will become `async def` and `yield from` will become `await`.